sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -2937,21 +2939,13 @@ public void testCountStarWithDegenerateFilter() throws Exception
                   .dataSource(CalciteTests.DATASOURCE1)
                   .intervals(querySegmentSpec(Filtration.eternity()))
                   .granularity(Granularities.ALL)
-                  .filters(
-                      and(
-                          selector("dim2", "a", null),
-                          or(
-                              bound("dim1", "a", null, true, false, null, StringComparators.LEXICOGRAPHIC),
-                              not(selector("dim1", null, null))
-                          )
-                      )
-                  )
+                  .filters(selector("dim2", "a", null))
```

Comment:
Here we have `dim2 = 'a' AND (dim1 > 'a' OR dim1 < 'b')` being translated as,

- Old: `dim2 = 'a' AND (dim1 > 'a' OR dim1 IS NOT NULL)`
- New: `dim2 = 'a'`

The new translation is equivalent, and better.